src/components/GasFeeWarning.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useEffect } from 'react'
+import { useTranslation } from 'react-i18next'
+import { StyleSheet } from 'react-native'
+import AppAnalytics from 'src/analytics/AppAnalytics'
+import { AppEvents } from 'src/analytics/Events'
+import InLineNotification, { NotificationVariant } from 'src/components/InLineNotification'
+import { Spacing } from 'src/styles/styles'
+import { PreparedTransactionsResult } from 'src/viem/prepareTransactions'
+
+export enum GasFeeWarningFlow {
+  Send = 'Send',
+  Swap = 'Swap',
+  Deposit = 'Deposit',
+  Withdraw = 'Withdraw',
+  Dapp = 'Dapp',
+}
+
+function GasFeeWarning({
+  prepareTransactionsResult,
+  flow,
+  onPressCta,
+  testIdPrefix,
+}: {
+  prepareTransactionsResult?: PreparedTransactionsResult
+  flow: GasFeeWarningFlow
+  onPressCta?: () => void
+  testIdPrefix?: string
+}) {
+  const { t } = useTranslation()
+
+  const flowToActionString = {
+    [GasFeeWarningFlow.Send]: t('gasFeeWarning.actions.sending'),
+    [GasFeeWarningFlow.Swap]: t('gasFeeWarning.actions.swapping'),
+    [GasFeeWarningFlow.Deposit]: t('gasFeeWarning.actions.depositing'),
+    [GasFeeWarningFlow.Withdraw]: t('gasFeeWarning.actions.withdrawing'),
+    [GasFeeWarningFlow.Dapp]: undefined,
+  }
+
+  const flowToVerbString = {
+    [GasFeeWarningFlow.Send]: t('gasFeeWarning.verb.send'),
+    [GasFeeWarningFlow.Swap]: t('gasFeeWarning.verb.swap'),
+    [GasFeeWarningFlow.Deposit]: t('gasFeeWarning.verb.deposit'),
+    [GasFeeWarningFlow.Withdraw]: t('gasFeeWarning.verb.withdraw'),
+    [GasFeeWarningFlow.Dapp]: undefined,
+  }
+
+  useEffect(() => {
+    if (prepareTransactionsResult && prepareTransactionsResult.type !== 'possible') {
+      AppAnalytics.track(AppEvents.show_gas_fee_warning, {
+        flow,
+        errorType: prepareTransactionsResult.type,
+        tokenNeeded: feeCurrencySymbol,
+      })
+    }
+  }, [prepareTransactionsResult])
+
+  if (!prepareTransactionsResult || prepareTransactionsResult.type === 'possible') {
+    return null
+  }
+
+  const feeCurrencySymbol =
+    prepareTransactionsResult.type === 'not-enough-balance-for-gas'
+      ? prepareTransactionsResult.feeCurrencies[0].symbol
```

Comment:
In the designs, it seems like it leans towards hardcoding this to be CELO or ETH, however in all of the use cases currently it is done like this. I think this way is easier for two reasons:
- ETH tokenId is not available in NetworkConfig, and would be needed for CTA, so would need to be added for all networks, increasing amount of hardcoded things
- The CTA is implemented in the component that uses this warning, so the implementer would have to get the hardcoded tokenId based on the network of the prepared transaction, rather than just looking at the first feeCurrencies.

I can add in the ETH tokenId's if we want but wanted to check first about doing it this way instead.